src/HealthChecks.Elasticsearch/DependencyInjection/ElasticsearchHealthCheckBuilderExtensions.cs

```diff
@@ -68,11 +67,56 @@ public static class ElasticsearchHealthCheckBuilderExtensions
 
         options.RequestTimeout ??= timeout;
 
+        if (options.Uri is null && !options.AuthenticateWithElasticCloud)
+        {
+            throw new InvalidOperationException($"there is no server to connect. consider using ${nameof(ElasticsearchOptions.UseElasticCloud)} or ${nameof(ElasticsearchOptions.UseServer)}");
+        }
+
         return builder.Add(new HealthCheckRegistration(
             name ?? NAME,
             sp => new ElasticsearchHealthCheck(options),
             failureStatus,
             tags,
             timeout));
     }
+
+    /// <summary>
+    /// Add a health check for Elasticsearch databases.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+    /// <param name="clientFactory">
+    /// An optional factory to obtain <see cref="Elastic.Clients.Elasticsearch.ElasticsearchClient" /> instance.
+    /// When not provided, <see cref="Elastic.Clients.Elasticsearch.ElasticsearchClient" /> is simply resolved from <see cref="IServiceProvider"/>.</param>
+    /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'elasticsearch' will be used for the name.</param>
+    /// <param name="failureStatus">
+    /// The <see cref="HealthStatus"/> that should be reported when the health check fails. Optional. If <c>null</c> then
+    /// the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
+    /// </param>
+    /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
+    /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
+    /// <returns>The specified <paramref name="builder"/>.</returns>
+    public static IHealthChecksBuilder AddElasticsearch(
+        this IHealthChecksBuilder builder,
+        Func<IServiceProvider, Elastic.Clients.Elasticsearch.ElasticsearchClient>? clientFactory = null,
+        string? name = default,
+        HealthStatus? failureStatus = default,
+        IEnumerable<string>? tags = default,
+        TimeSpan? timeout = default)
+    {
+
+        return builder.Add(new HealthCheckRegistration(
+            name ?? NAME,
+            sp =>
+            {
+                var options = new ElasticsearchOptions();
+                options.RequestTimeout = timeout;
+
+                options.Client = clientFactory?.Invoke(sp) ?? sp.GetRequiredService<Elastic.Clients.Elasticsearch.ElasticsearchClient>();

```

Comment:
Let's hope that this is going to solve "Object initialization can be simplified"

```suggestion
                ElasticsearchOptions options = new()
                {
                    RequestTimeout = timeout,
                    Client = clientFactory?.Invoke(sp) ?? sp.GetRequiredService<Elastic.Clients.Elasticsearch.ElasticsearchClient>()
                };

```